src/main/webapp/script.js

```diff
@@ -36,4 +36,124 @@ function displayError() {
   // TODO: Write error to user interface
   console.log('Geolocation not enabled');
   alert(alertMessage);
+}
+
+//Array of the (currently 6 for this demo build) 15 listings gathered from the fetch request
+const listingsArray = [];
+let count = 0;
+
+function displayList() {
+  fetch('/data').then(response => response.json()).then((listings) => {
+    listings.forEach((listing) => {
+      listingsArray.push(createResultCard(listing.name, listing.address, listing.image, listing.rating, listing.website, count));
+      count++;
+    });
+    limitList('begin');
+  });
+}
+
+//Variable keeping track of the card that is currently shown at the bottom of the list
+let currentEndCard;
+
+/**
+ * @param {string} direction Contains which direction the list should iterate
+ */
+function limitList(direction) {
+  const resultsContent = document.getElementById('results-content');
+  resultsContent.className = 'results-content';
+
+  // TODO: Add better checking for end / beginning of list and change/remove alert
+  // TODO: Also add changing nav buttons based on this checking
+  if((currentEndCard - 2) == 0 && direction == 'previous') {
+    alert('Already at beginning of list!');
+  }
+  else if ((currentEndCard + 1) == count && direction == 'next') {
+    alert('Already at end of list!');
+  }
+  else {
+    resultsContent.innerHTML = '';
+    if(direction == 'begin') {
+      for (i = 0; i < 3; i++) {
```

Comment:
Do you declare 'i' anywhere? Not sure if that will cause issues with the javascript down the road

Reply:
Oops haha - dunno how that was working